plugins/modules/openssh_keypair.py

```diff
@@ -180,6 +204,39 @@ def __init__(self, module):
         if self.regenerate == 'always':
             self.force = True
 
+        # The empty string is intentionally ignored so that dependency checks do not cause unnecessary failure
+        if self.passphrase:
+            if not HAS_OPENSSH_SUPPORT:
+                module.fail_json(
+                    msg=missing_required_lib(
+                        'cryptography >= 2.6',
+                        reason="to encrypt/decrypt private keys with passphrases"
+                    )
+                )
+
+            ssh = module.get_bin_path('ssh', True)
+            proc = module.run_command([ssh, '-Vq'])
+            ssh_version = parse_openssh_version(proc[2].strip())
+            if LooseVersion(ssh_version) >= LooseVersion("7.8") and not HAS_OPENSSH_PRIVATE_FORMAT:
```

Comment:
How about adding another option for determining the key format? It should also have an `auto` value (which would be the default, and do what exactly you are doing right now) which determines the format from `ssh -Vq`.

In general you want (for example) to be able to force the old format even if you have a newer SSH installed locally, since you plan to copy the key to a remote machine which has an older version of SSH. In that case, being able to specify the format is pretty important.

Reply:
I can do this, but I assume you would also like me to implement for the no `passphrase` scenario via `-m {PEM|SSH}`?
Just confirming before I take a crack at it.

Reply:
I think for now I'd implement it only when `passphrase` is selected, and make it error out if set to any other value than `auto`. Implementing it for when ssh-keygen is used can be done in a follow-up PR (if you want, or someone else can do it when they want, or just implement it for the `cryptography` backend).

Reply:
Ok, I think everything is addressed, but I updated the 'PEM' format to be 'PKCS1' which I don't think is technically accurate, but fits the OpenSSH terminology. Cryptography just uses the name 'TraditionalOpenSSL' for the serialization format which uses the appropriate format for RSA, DSA, and ECDSA. Apparently ED25519 private keys are always in OpenSSH format even for newer versions of OpenSSH which support PKCS8 formatted keys so I added logic to check for this.